### PR TITLE
Refactoring parametric schedule code

### PR DESCRIPTION
**Pull request by TMaile**

### CODE REVIEW

File: lib/buildingsync/model_articulation/parametric_schedule_helper.rb [hunk 1]
```diff
@@ -0,0 +1,567 @@
...
+      create_schedule_occupancy(model, default_schedule_set, occupancy_profiles, profile_override, hoo_start_wkdy, hoo_end_wkdy, hoo_start_sat, hoo_end_sat, hoo_start_sun, hoo_end_sun)
+      create_schedule_infiltration(model, default_schedule_set, infiltration_profiles, profile_override, hoo_start_wkdy, hoo_end_wkdy, hoo_start_sat, hoo_end_sat, hoo_start_sun, hoo_end_sun)
+      create_schedule_hvac_availability(model, default_schedule_set, hvac_availability_profiles, profile_override, hoo_start_wkdy, hoo_end_wkdy, hoo_start_sat, hoo_end_sat, hoo_start_sun, hoo_end_sun)
+      create_schedule_heating_cooling(model, default_schedule_set, thermostat_setback_profiles, htg_setpoint, clg_setpoint, setback_delta, profile_override,hoo_start_wkdy, hoo_end_wkdy, hoo_start_sat, hoo_end_sat, hoo_start_sun, hoo_end_sun)
+      create_schedule_SHW(model, default_schedule_set, swh_profiles, profile_override, hoo_start_wkdy, hoo_end_wkdy, hoo_start_sat, hoo_end_sat, hoo_start_sun, hoo_end_sun)
+    end
+
+    def self.get_profiles(lighting_profiles = nil, electric_equipment_profiles = nil, gas_equipment_profiles = nil, occupancy_profiles = nil,
```
Comment (corymosiman12) [inline lib/buildingsync/model_articulation/parametric_schedule_helper.rb]: Please define these in an external data structure to read in.

File: lib/buildingsync/model_articulation/parametric_schedule_helper.rb [hunk 2]
```diff
@@ -0,0 +1,567 @@
+# *******************************************************************************
```
Comment (corymosiman12) [inline lib/buildingsync/model_articulation/parametric_schedule_helper.rb]: move this file under the `helpers` directory

File: lib/buildingsync/model_articulation/loads_system.rb [hunk 3]
```diff
@@ -148,60 +150,12 @@ def get_building_section(building_sections, standard_building_type, standard_spa
 
     def adjust_people_schedule(space_type, building_section, model)
```
Comment (corymosiman12) [inline lib/buildingsync/model_articulation/loads_system.rb]: The implementation of the `adjust_people_schedule` function uses the `get_building_section` function, which only returns a single building_section:
https://github.com/BuildingSync/BuildingSync-gem/blob/865aaa7164003d753a6e77c7838afffdbd1e4637/lib/buildingsync/model_articulation/loads_system.rb#L77

Please describe how this should / does work for multiple sections with different typical occupant hours.

**Review by corymosiman12 — CHANGES_REQUESTED** [review]
Add tests to show expected functionality for how the `typical_occupant_usage` values are intended to affect: activity, lighting, electrical_equipment ... etc.

The purpose is to illustrate through testing what the schedules look like before parametric schedules are applied and after.

Also, for the default schedules, please add some documentation (README, comments, etc.) describing where the default, weekend, and weekday values are coming from.

File: lib/buildingsync/model_articulation/parametric_schedule_helper.rb (hunk 2 above)
Comment (TMaile) [reply lib/buildingsync/model_articulation/parametric_schedule_helper.rb]: done here https://github.com/BuildingSync/BuildingSync-gem/commit/ad0a657dda8010b4bb04cb23cfc4dd22d92ed8a3

File: lib/buildingsync/model_articulation/parametric_schedule_helper.rb (hunk 1 above)
Comment (TMaile) [reply lib/buildingsync/model_articulation/parametric_schedule_helper.rb]: profiles are saved in a hash and hours of operation in a new helper class. see https://github.com/BuildingSync/BuildingSync-gem/commit/3316472d9aad81b329857f3f225fd0efee051d25

File: lib/buildingsync/model_articulation/loads_system.rb (hunk 3 above)
Comment (TMaile) [reply lib/buildingsync/model_articulation/loads_system.rb]: We iterate over all space types and adjust the related schedules in each schedule set of each space type. 
The building section is needed to get the parameter hours_per_week that is defined at that level.

**Comment by TMaile** [comment]
The schedule adjustment are for all schedules that are attached to a default schedule set. 
I added test code to verify that the schedule hours above 0.5 are equal to a specific value. 
The default schedules are generated with the Standards Gem based on the building type and standard used.

**Comment by TMaile** [comment]
Just to clarify, the related measure and this current code, do not adjust the existing schedules, they create new schedules based hours per week and default start and end times.

**Comment by TMaile** [comment]
@corymosiman12  can you confirm that you are ok with these changes? (right now this is the last branch not merged yet)

**Review by corymosiman12 — CHANGES_REQUESTED** [review]
Hey @TMaile -

Ok so had a quick conversation with David.  I guess some of this parametric schedules stuff is already in the Standards library.  Would you take a look at those methods and see if they accomplish what we need them to?  I personally would prefer using the stuff from the Standards gem if possible.  Sorry for not knowing about this sooner.

Screen Shot 2020-10-29 at 5 26 55 PM

**Comment by TMaile** [comment]
Adjusted the code to use the parametric schedule code within the Standards library. 
The limitation right now is that this adjust the schedules for the complete model and we cannot apply changes only on a space type level.

File: lib/buildingsync/model_articulation/loads_system.rb [hunk 4]
```diff
@@ -149,26 +149,59 @@ def get_building_section(building_sections, standard_building_type, standard_spa
       return nil
     end
 
-    def adjust_people_schedule(space_type, building_section, model)
-      if !building_section.typical_occupant_usage_value_hours.nil?
-        puts_string = "building_section.typical_occupant_usage_value_hours: #{building_section.typical_occupant_usage_value_hours}"
+    def adjust_schedules(standard, space_type, building_section, model)
```
Comment (corymosiman12) [inline lib/buildingsync/model_articulation/loads_system.rb]: @TMaile is this a `building_section` or more at the `building` level?  Please change parameter name to `building` if so

**Review by corymosiman12 — CHANGES_REQUESTED** [review]
@TMaile minor change requested - else looks good!

File: lib/buildingsync/model_articulation/loads_system.rb (hunk 4 above)
Comment (TMaile) [reply lib/buildingsync/model_articulation/loads_system.rb]: this is addressed in https://github.com/BuildingSync/BuildingSync-gem/commit/1d14b861aad9edc057934b64af8d45f50ee45cd9
taking the average value from the sections at the building level

**Comment by TMaile** [comment]
extended documentation to include the last classes and functions for system type derivatives

**Pull request merged**